src/classes/RD2_ApiService.cls

```diff
@@ -253,4 +258,287 @@ public inherited sharing class RD2_ApiService {
+        public String error;
+
+        public PauseObject addError(String errorString) {
+            this.isSuccess = false;
+            this.error = errorString;
+
+            return this;
+        }
+
+        public PauseObject populateDefaultValues() {
+            this.isSuccess = true;
+            this.unPause = (this.unPause == null)
+                ? false
+                : this.unPause;
+
+            return this;
+        }
+    }
+
+    public class PauseValidator {
+        PauseObject pause;
+
+        PauseValidator(PauseObject pause) {
+            this.pause = pause;
+        }
+
+        public void validatePauseData(Id rdId, RD2_RecurringDonation recurringDonation) {
+            validateRecurringDonationId(rdId);
+            validateStateShouldBeActive(recurringDonation);
+            validateRecurringDonationNotLinkedToElevate(recurringDonation);
+
+            if (this.pause.unPause) {
+                validateUnPause();
+            } else {
+                validateDateRange();
+            }
+        }
+
+        private void validateRecurringDonationId(Id recordId) {
+            if (UTIL_Describe.isObjectIdThisType(recordId, 'npe03__Recurring_Donation__c') == false) {
+                pause.addError(String.format(System.Label.RD2_InvalidRecurringDonationId,
+                        new List<String> { recordId }));
+            }
+        }
+
+        private void validateStateShouldBeActive(RD2_RecurringDonation recurringDonation) {
+            if (!pause.isSuccess) {
+                return;
+            }
+
+            if (!recurringDonation.isActive()) {
+                pause.addError(System.Label.RD2_StatusMustBeActiveOrPause);
+            }
+        }
+
+        private void validateRecurringDonationNotLinkedToElevate(RD2_RecurringDonation recurringDonation) {
+            if (!pause.isSuccess) {
+                return;
+            }
+
+            if (recurringDonation.isElevateRecord()) {
+                pause.addError(System.Label.RD2_ElevateRDCannotBePaused);
+            }
+        }
+
+        private void validateUnPause() {
+            if (!pause.isSuccess) {
+                return;
+            }
+
+            if (pause.startDate != null || pause.endDate != null) {
+                pause.addError(System.Label.RD2_UnPauseRequestCannotIncludeDates);
+            }
+        }
+
+        private void validateEndDateGreaterThanOrEqualToStartDate() {
+            if (!pause.isSuccess) {
+                return;
+            }
+
+            if (pause.startDate > pause.endDate) {
+                pause.addError(System.Label.RD2_PauseEndDateMustBeGreaterThanStartDate);
+            }
+        }
+
+        private void validateDateRange() {
+            validateDatesArePopulated();
+            validateEndDateGreaterThanOrEqualToStartDate();
+            validateStartDateGreaterThanOrEqualToToday();
+            validateEndDateNoGreaterThanTwelveMonthsFromStartDate();
+        }
```

Comment:
This seems unnecessary. The code on line 456 could just call these 4 methods rather than this one method that calls the four methods. Plus, at some point it may be necessary to validate something beyond just dates.